File: MyAPI/MyAPI/Data/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MyAPI.Data
{
    public class Order
    {
        public int Id { get; set; }
        [ForeignKey(nameof(UserID))]
        public string UserID { get; set; }
        public virtual APIUser User { get; set; }
        public string ContactName { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string PaymentMethod { get; set; }
        public string OrderDate { get; set; }
        public int TotalItem { get; set; }
        public double TotalPrice { get; set; }

        public int Status { get; set; }
        public string Note { get; set; }
        public virtual IList<OrderDetail> OrderDetails { get; set; }
    }
}

File: MyFE/src/app/class/order.ts
import { OrderDetail } from "./order-detail"
import { User } from "./user"

export class Order {
    id!:number
    user!:User
    userID!:string
    contactName!:string
    address!:string
    phone!: string
    email!: string
    orderDate!:string
    paymentMethod!: string
    status!: number
    totalItem!:number
    totalPrice!:number
    shipper!:User
    note!:string
    orderDetails!:OrderDetail[]
}
